graphene_parser.py

```diff
@@ -517,6 +521,20 @@ def try_parse_fn_from_specialization(
                 resolved_return = self._program.resolve_type(
                     unresolved_return.produce_specialized_copy(specialization_map)
                 )
+            except SubstitutionFailure:
+                return None  # SFINAE
```

Comment:
Wait... How did this work before?

Reply:
We were always catching `SubstitutionFailure`s (now lines 553-554). I had to split the try/catch such that we can attach the location information of the return type specifically.